src/screens/LoginScreen.tsx

```diff
@@ -51,57 +60,65 @@ export default function LoginScreen({navigation}): JSX.Element {
         blurRadius={10}
         source={require('../../assets/images/logo_taskratchet_square_64@2.png')}
       />
-      <SafeAreaView style={styles.container}>
-        <Image
-          style={{width: 60, height: 60}}
-          source={require('../../assets/images/logo_taskratchet_512_bordered.png')}
-        />
-        <View style={styles.titleGroup}>
-          <Text style={[textColorStyle, styles.title]}>TaskRatchet</Text>
-          <Text style={[textColorStyle, styles.title]}>Login</Text>
-        </View>
-
-        <View style={styles.credentials}>
-          <View style={styles.inputGroup}>
-            <Text style={[textColorStyle, styles.inputTitle]}>Username</Text>
-            <TextInput
-              style={[textColorStyle, styles.inputField]}
-              onChangeText={setUserInput}
-              placeholder="Username"
-              placeholderTextColor={useIsDarkMode() ? 'white' : 'black'}
-              keyboardType="default"
-            />
+      <SafeAreaView>
+        <KeyboardAvoidingView style={styles.container}>
+          <Image
+            style={{width: 60, height: 60}}
+            source={require('../../assets/images/logo_taskratchet_512_bordered.png')}
+          />
+          <View style={styles.titleGroup}>
+            <Text style={[textColorStyle, styles.title]}>TaskRatchet</Text>
+            <Text style={[textColorStyle, styles.title]}>Login</Text>
           </View>
-          <View style={styles.inputGroup}>
-            <Text style={[textColorStyle, styles.inputTitle]}>Password</Text>
-            <TextInput
-              style={[textColorStyle, styles.inputField]}
-              onChangeText={setPassInput}
-              placeholder="Password"
-              placeholderTextColor={useIsDarkMode() ? 'white' : 'black'}
-              keyboardType="default"
-            />
+
+          <View style={styles.credentials}>
+            <View style={styles.inputGroup}>
+              <Text style={[textColorStyle, styles.inputTitle]}>Username</Text>
+              <TextInput
+                style={[textColorStyle, styles.inputField]}
+                onChangeText={setUserInput}
+                placeholder="Username"
+                placeholderTextColor={useIsDarkMode() ? 'white' : 'black'}
```

Comment:
Extract `const isDarkMode = useIsDarkMode()` to avoid multiple calls?

Reply:
I just went ahead and went through all the code extracting any place with useIsDarkMode into a separate const for the sake of efficiency.